File: Components/WinForms/VB/EditorsMainDemo/Modules/ModuleFontEdit.vb
Imports System.Collections
Imports System.ComponentModel
' <ceShowCustomButtons>
Imports DevExpress.XtraEditors.Controls
' </ceShowCustomButtons>
Imports DevExpress.Tutorials


Namespace DevExpress.XtraEditors.Demos
	Partial Public Class ModuleFontEdit
		Inherits TutorialControl

		Public Sub New()
			InitializeComponent()
			TutorialInfo.WhatsThisCodeFile = "VB\EditorsMainDemo\Modules\ModuleFontEdit.vb"
			TutorialInfo.WhatsThisXMLFile = "fontEdit.xml"
		End Sub

		Private updateValues As Boolean = False

		Private Sub ModuleCalcEdit_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
			ControlUtils.CenterControlInParent(fontEditSample)
			InitValues()
			fontEditSample.EditValue = Me.Font.FontFamily.GetName(0)
			cbFontSize.EditValue = fontEditSample.Properties.AppearanceDropDown.Font.Size.ToString()
		End Sub

		Private Sub InitValues()
			updateValues = True
			seRUItemCount.Value = fontEditSample.Properties.RecentlyUsedItemCount
			ceShowSymbolPreview.Checked = fontEditSample.Properties.ShowSymbolFontPreview
			updateValues = False
		End Sub

		'<seRUItemCount>
		Private Sub seRUItemCount_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles seRUItemCount.EditValueChanged
			If updateValues Then
				Return
			End If
			fontEditSample.Properties.RecentlyUsedItemCount = Convert.ToInt32(seRUItemCount.Value)
		End Sub
		'</seRUItemCount>

		'<ceShowSymbolPreview>
		Private Sub ceShowSymbolPreview_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles ceShowSymbolPreview.CheckedChanged
			If updateValues Then
				Return
			End If
			fontEditSample.Properties.ShowSymbolFontPreview = ceShowSymbolPreview.Checked
		End Sub

		Private Sub cbFontSize_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbFontSize.SelectedIndexChanged
			If updateValues Then
				Return
			End If
			fontEditSample.Properties.AppearanceDropDown.Font = New Font(DefaultFont.Name, Convert.ToInt32(cbFontSize.EditValue))
		End Sub
		'</ceShowSymbolPreview>
	End Class
End Namespace

File: Components/WinForms/CS/EditorsMainDemo/Modules/ModuleFontEdit.cs
using System;
using System.Collections;
using System.ComponentModel;
using System.Drawing;
// <ceShowCustomButtons>
using DevExpress.XtraEditors.Controls;
// </ceShowCustomButtons>
using System.Windows.Forms;
using DevExpress.Tutorials;


namespace DevExpress.XtraEditors.Demos {
    public partial class ModuleFontEdit : TutorialControl  {
        public ModuleFontEdit() {
            InitializeComponent();
            TutorialInfo.WhatsThisCodeFile = "CS\\EditorsMainDemo\\Modules\\ModuleFontEdit.cs";
            TutorialInfo.WhatsThisXMLFile = "DevExpress.XtraEditors.Demos.CodeInfo.fontEdit.xml";
        }
        protected override void UpdateTabControlPosition(XtraTab.XtraTabControl tabControl) { }
        protected override void CalcContentCore() {
            CalcLayoutControlBounds(layoutControl1);
        }

        bool updateValues = false;

        private void ModuleCalcEdit_Load(object sender, System.EventArgs e) {
            ControlUtils.CenterControlInParent(fontEditSample);
            InitValues();
            fontEditSample.EditValue = this.Font.FontFamily.GetName(0);
            cbFontSize.EditValue = fontEditSample.Properties.AppearanceDropDown.Font.Size.ToString(); 
        }

        void InitValues() {
            updateValues = true;
            seRUItemCount.Value = fontEditSample.Properties.RecentlyUsedItemCount;
            ceShowSymbolPreview.Checked = fontEditSample.Properties.ShowSymbolFontPreview;
            updateValues = false;
        }

        //<seRUItemCount>
        private void seRUItemCount_EditValueChanged(object sender, EventArgs e) {
            if(updateValues) return;
            fontEditSample.Properties.RecentlyUsedItemCount = Convert.ToInt32(seRUItemCount.Value);
        }
        //</seRUItemCount>

        //<ceShowSymbolPreview>
        private void ceShowSymbolPreview_CheckedChanged(object sender, EventArgs e) {
            if(updateValues) return;
            fontEditSample.Properties.ShowSymbolFontPreview = ceShowSymbolPreview.Checked;
        }

        private void cbFontSize_SelectedIndexChanged(object sender, EventArgs e) {
            if(updateValues) return;
            fontEditSample.Properties.AppearanceDropDown.Font = new Font(DefaultFont.Name, Convert.ToInt32(cbFontSize.EditValue));  
        }
        //</ceShowSymbolPreview>
    }
}

